app/models/jobstatusdata.rb

```diff
@@ -7,77 +7,146 @@
 
   # Define an object containing only necessary data to send to client.
   #
   # Object defaults to condensed data, add extended flag to initializer to include all data used by the application.
   #
   # @return [Jobstatusdata] self
-  def initialize(pbs_job, pbs_cluster=Servers.first[0], extended=false)
-    self.pbsid = pbs_job[:name]
-    self.jobname = pbs_job[:attribs][:Job_Name]
-    self.username = username_format(pbs_job[:attribs][:Job_Owner])
-    self.account = pbs_job[:attribs][:Account_Name]
-    self.status = pbs_job[:attribs][:job_state]
-    if self.status == "R" || self.status == "C"
-      self.nodes = node_array(pbs_job[:attribs][:exec_host])
-      self.starttime = pbs_job[:attribs][:start_time]
+  def initialize(info, cluster=OODClusters.first.id.to_s, extended=false)
+    self.pbsid = info.id
+    self.jobname = info.job_name
+    self.username = info.job_owner
+    self.account = info.accounting_id
+    self.status = info.status.state
+    self.cluster = cluster
+    self.walltime_used = info.wallclock_time > 0 ? pretty_time(info.wallclock_time) : ''
+    self.queue = info.queue_name
+    if info.status == :running || info.status == :completed
+      self.nodes = node_array(info.allocated_nodes)
+      self.starttime = info.dispatch_time.to_i || 0
     end
-    self.cluster = pbs_cluster
+    # TODO Find a better way to distingush whether a native parser is available. Maybe this is fine?
+    self.extended_available = OODClusters[cluster].job_config[:adapter] == "torque" || OODClusters[cluster].job_config[:adapter] == "slurm"
     if extended
-      extended_data(pbs_job)
+      if OODClusters[cluster].job_config[:adapter] == "torque"
+        extended_data_torque(info)
+      elsif OODClusters[cluster].job_config[:adapter] == "slurm"
+        extended_data_slurm(info)
+      else
+        extended_data_default(info)
+      end
     end
     self
   end
 
-  # Store additional data about the job.
+  # Store additional data about the job. (Torque-specific)
+  #
+  # Parses the `native` info function for additional information about jobs on Torque systems.
   #
   # @return [Jobstatusdata] self
-  def extended_data(pbs_job)
-    self.walltime = pbs_job[:attribs][:Resource_List][:walltime]
-    self.walltime_used = pbs_job[:attribs].fetch(:resources_used, {})[:walltime].presence || 0
-    self.submit_args = pbs_job[:attribs][:submit_args].presence || "None"
-    self.output_path = pbs_job[:attribs][:Output_Path].to_s.split(":").second || pbs_job[:attribs][:Output_Path]
-    self.nodect = pbs_job[:attribs][:Resource_List][:nodect]
-    self.ppn = pbs_job[:attribs][:Resource_List][:nodes].to_s.split("ppn=").second || 0
+  def extended_data_torque(info)
+    self.walltime = info.native[:Resource_List][:walltime]
```

Comment:
Like `info.native.fetch(:resources_used, {})[:cput].presence` below, something similar whenever you use `info.native[:Resource_List]` instead of assuming its always there?